packages/video_player_avplayer/tizen/src/plus_player.cc

```diff
@@ -0,0 +1,608 @@
+// Copyright 2023 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "plus_player.h"
+
+#include <app_manager.h>
+#include <dlfcn.h>
+#include <system_info.h>
+#include <unistd.h>
+
+#include "log.h"
+
+PlusPlayer::PlusPlayer(flutter::BinaryMessenger *messenger, void *native_window,
+                       std::string &video_format)
+    : VideoPlayer(messenger),
+      native_window_(native_window),
+      video_format_(video_format) {}
+
+PlusPlayer::~PlusPlayer() { Dispose(); }
+
+int64_t PlusPlayer::Create(const std::string &uri, int drm_type,
+                           const std::string &license_server_url,
+                           bool is_prebuffer_mode) {
+  LOG_INFO("[PlusPlayer] Create player.");
+  if (!video_format_.compare("dash")) {
+    player_ = plusplayer::PlusPlayer::Create(plusplayer::PlayerType::kDASH);
+  } else {
+    player_ = plusplayer::PlusPlayer::Create();
+  }
+
+  if (!player_) {
+    LOG_ERROR("[PlusPlayer] Fail to create player.");
+    return -1;
+  }
+
+  if (!player_->Open(uri)) {
+    LOG_ERROR("[PlusPlayer] Fail to open uri :  %s.", uri.c_str());
+    return -1;
+  }
+
+  char *appId = nullptr;
+  int ret = app_manager_get_app_id(getpid(), &appId);
+  if (ret != APP_MANAGER_ERROR_NONE) {
+    LOG_ERROR("[PlusPlayer] app_manager_get_app_id failed : %s.",
+              get_error_message(ret));
+    return -1;
+  }
+  player_->SetAppId(std::string(appId));
+  free(appId);
+
+  player_->RegisterListener(this);
+
+  if (drm_type != 0) {
+    if (!SetDrm(uri, drm_type, license_server_url)) {
+      LOG_ERROR("[PlusPlayer] Failed to set drm.");
+      return -1;
+    }
+  }
+
+  if (!SetDisplay()) {
+    LOG_ERROR("[PlusPlayer] Failed to set display.");
+    return -1;
+  }
+
+  SetDisplayRoi(0, 0, 1, 1);
+
+  if (is_prebuffer_mode) {
+    player_->SetPrebufferMode(true);
+    is_prebuffer_mode_ = true;
+  }
+
+  if (!player_->PrepareAsync()) {
+    LOG_ERROR("[PlusPlayer] Failed to prepare");
+    return -1;
+  }
+  return SetUpEventChannel();
+}
+
+void PlusPlayer::Dispose() {
+  LOG_INFO("[PlusPlayer] Disposing.");
+
+  if (player_) {
+    player_->Stop();
+    player_->Close();
```

Comment:
According to the document, a state check is needed before calling Close(). (Same applies to Stop())
```
   * @pre       The player state must be one of
   *            State::kIdle or State::kNone
```

Reply:
Fixed.
[Codes cleanup](https://github.com/xiaowei-guan/plugins/pull/6/commits/5f09b1831625607e9677d98d35788ac9db34b2c7)